readme.md

```diff
@@ -1,3 +1,53 @@
 # Weebly Client
 
-Soon to be a client for using the weebly platform API
\ No newline at end of file
+Weebly Client is a basic PHP class that simplifies interactions with the Weebly REST API.
+
+You can find more details about how to get set-up with your Weebly project at <https://dev.weebly.com/get-started-with-developing-apps.html>.
+
+You can find details about the various endpoints that you may access with the Weebly Client at <https://dev.weebly.com/about-rest-apis.html>.
+
+## Usage:
+
+### Instantiating the Client
+
+You may insantiate a new Weebly Client as such:
+
+```
+$client = \Weebly\WeeblyClient($client_id, $client_secret, $user_id, $site_id, $access_token);
+```
+
+The only mandatory options that you must provide to the constructor are `$client_id` and `$client_secret`. You can find them both in your dev dashboard at <https://www.weebly.com/developer-admin> once you have registered a developer account with Weebly.
+Please store these credentials securely on your server, and not expose them publicly.
+The `$user_id`, `site_id` and `$access_token` are made accessible to you when a user grants your app permissions through the App OAuth flow. You do not need to provide these parameters unless you are accessing the OAuth methods. (i.e. _*Store APIs*_, _*Site APIs*_)
+You can find more information about the OAuth flow and how to obtain the `$user_id`, `site_id` and `$access_token` at <https://dev.weebly.com/configure-oauth.html>.
+
+### Authorizing through the OAuth Flow
+The WeeblyClient provides a few helper methods to aid you through the OAuth flow.
+    * `getAuthorizationUrl` Returns a properly formatted authorization URL that you should respond with after our servers initiate the OAuth flow.
+    * `getAccessToken` :Retrieves the access token once you have obtained the necessary authorization code in OAuth.
+
+You can find more information about the Weebly OAuth flow at <https://dev.weebly.com/configure-oauth.html>.
```

Comment:
You may want to consider pointing them to <https://dev.weebly.com/authentication-with-oauth2.html> first, so they understand the flow. It's kind of backwards from most OAuth flows so if they go right into config, they might get confused.